addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/handler/NeatoHandler.java

```diff
@@ -0,0 +1,255 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neato.handler;
+
+import static org.openhab.binding.neato.NeatoBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.neato.internal.NeatoRobot;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NeatoHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Patrik Wimnell - Initial contribution
+ */
+public class NeatoHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(NeatoHandler.class);
+
+    private String vacuumSerialNumber;
+    private String vacuumSecret;
+    private String vacuumName;
+
+    private NeatoRobot mrRobot;
+
+    private BigDecimal refreshTime;
+    private ScheduledFuture<?> refreshTask;
+    private static BigDecimal DEFAULTREFRESHTIME = new BigDecimal(60.0);
+
+    public NeatoHandler(Thing thing) {
+        super(thing);
+    }
+
+    public ScheduledFuture<?> getRefreshTask() {
+        return this.refreshTask;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(org.openhab.binding.neato.NeatoBindingConstants.COMMAND)) {
+            logger.debug("Ok - will handle command for CHANNEL_COMMAND");
+            // TODO: handle command
+
+            try {
+                mrRobot.sendCommand(command);
+            } catch (Exception exc) {
+
+            }
+            // Note: if communication with thing fails for some reason,
+            // indicate that by setting the status with detail information
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+            // "Could not control device at IP address x.x.x.x");
+
+        }
+        this.refreshStateAndUpdate();
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.info("Will boot up Neato Vacuum Cleaner binding!");
+        // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
+
+        Configuration config = getThing().getConfiguration();
```

Comment:
So for you information, framework support also transforming configuration directly to java object. Just define configuration class which can hold all of you configuration parameters. This simplifies configuration handling a lot as you don't need to check every parameter manually in the code (framework do that for you against thing configuration description file.